src/app/views/contact/contact.component.html

```diff
@@ -0,0 +1,48 @@
+<div class="container-fluid">
+
+  <div class="contact-header row">
+    <!-- All Questions -->
+    <div class="col-xs-12">
+      <h3>
+        {{"contact" | translate}}
+      </h3>
+    </div>
+  </div>
+
+  <div class="contact-form-container">
+    <div class="textarea">
+      <p>
+        {{"contact.description" | translate}} {{"contact.instruction" | translate}}
+      </p>
+    </div>
+
+    <div class="envelope-icon-container">
+      <i class="fa fa-envelope fa-5x" aria-hidden="true"></i>
+    </div>
+
+    <form [formGroup]="contactForm" (ngSubmit)="submitForm(contactForm.value)">
+      <div class="form-group material-group contact-input-container" [ngClass]="{'has-error':!contactForm.controls['name'].valid && contactForm.controls['name'].touched}">
+        <input class="form-control material-input" type="text" placeholder="{{'contact.form.name' | translate}}" [formControl]="contactForm.controls['name']">
+
+        <div *ngIf="contactForm.controls['name'].hasError('required') && contactForm.controls['name'].touched" class="alert alert-danger">{{'contact.form.name.validation' | translate}}.</div>
+        <div *ngIf="contactForm.controls['name'].hasError('minlength') && contactForm.controls['name'].touched" class="alert alert-danger">{{'contact.form.name.validation.length' | translate}}.</div>
+      </div>
+      <div class="form-group  material-group contact-input-container" [ngClass]="{'has-error':!contactForm.controls['email'].valid && contactForm.controls['email'].touched}">
+        <input class="form-control material-input" type="email" placeholder="{{'contact.form.email' | translate}}" [formControl]="contactForm.controls['email']">
+
+        <div *ngIf="contactForm.controls['email'].hasError('required') && contactForm.controls['email'].touched" class="alert alert-danger">{{'contact.form.email.validation' | translate}}.</div>
+        <div *ngIf="contactForm.controls['email'].hasError('minlength') && contactForm.controls['email'].touched" class="alert alert-danger">{{'contact.form.email.validation.length' | translate}}.</div>
+      </div>
+      <div class="form-group  material-group contact-input-container" [ngClass]="{'has-error':!contactForm.controls['message'].valid && contactForm.controls['message'].touched}">
+        <input class="form-control material-input" type="text" placeholder="{{'contact.form.message' | translate}}" [formControl]="contactForm.controls['message']">
+
+        <div *ngIf="contactForm.controls['message'].hasError('required') && contactForm.controls['message'].touched" class="alert alert-danger">{{'contact.form.message.validation' | translate}}.</div>
+        <div *ngIf="contactForm.controls['message'].hasError('minlength') && contactForm.controls['message'].touched" class="alert alert-danger">{{'contact.form.message.validation.length' | translate}}.</div>
```

Comment:
This is a bit out of scope, but think no is a good time to ask :):

I'm not sure if ng2 has a convention or but this is a pattern we seem to repeat quite a lot. Is there some abstraction we can provide for this?
This is me speculating, but perhaps something like an `InputError` component?

It would be nice to have something like it since we could then add more advanced error rules in the future. One thing I've implemented before which worked great was to later on add a concept of "soft validation" errors.

@bex1, @bvangelov what do you think? :)